node/src/reactor/initializer.rs

```diff
@@ -93,45 +95,17 @@ impl From<StorageRequest> for Event {
     }
 }
 
-impl From<NetworkRequest<Message>> for Event {
-    fn from(_request: NetworkRequest<Message>) -> Self {
-        unreachable!("no network traffic happens during initialization")
-    }
-}
-
 impl From<ChainspecLoaderAnnouncement> for Event {
-    fn from(_announcement: ChainspecLoaderAnnouncement) -> Self {
-        unreachable!("no chainspec announcements happen during initialization")
-    }
-}
-
-impl From<LinearChainRequest> for Event {
-    fn from(_req: LinearChainRequest) -> Self {
-        unreachable!("no linear chain events happen during initialization")
-    }
-}
-
-impl From<NetworkRequest<gossiper::Message<GossipedAddress>>> for Event {
-    fn from(_request: NetworkRequest<gossiper::Message<GossipedAddress>>) -> Self {
-        unreachable!("no gossiper events happen during initialization")
-    }
-}
-
-impl From<ConsensusRequest> for Event {
-    fn from(_request: ConsensusRequest) -> Self {
-        unreachable!("no chainspec announcements happen during initialization")
-    }
-}
-
-impl From<RestRequest> for Event {
-    fn from(_request: RestRequest) -> Self {
-        unreachable!("no rest requests happen during initialization")
+    fn from(announcement: ChainspecLoaderAnnouncement) -> Self {
+        info!(%announcement, "chainspec announcements received by initializer, ignoring");
+        Event::ChainspecLoaderAnnouncement
     }
 }
 
 impl From<ContractRuntimeAnnouncement> for Event {
```

Comment:
I think we can just derive this using `#[from]` instead, right? I'd upgrade the `info!` message that handles it to a `warn!` and drop the `error!` here.

Reply:
Auto derive will require extending the enum variant, won't it?

Like so:
```
ChainspecLoaderAnnouncement(ChainspecLoaderAnnouncement),
```

I'd like to avoid it, since we don't need the inner `ChainspecLoaderAnnouncement`.

Reply:
> I'd upgrade the `info!` message that handles it to a `warn!` and drop the `error!` here.

`info` level was explicitly mentioned here. I think it's ok since ignoring the `ChainspecLoaderAnnouncement` can do no harm to the system.

Reply:
As agreed during the call, the changes are introduced here: https://github.com/casper-network/casper-node/pull/2700/commits/3b8cbd4b058fb3f79d57362fd7567d7d29f9a1b9